src/main/scala/is/hail/variant/VariantDataset.scala

```diff
@@ -1178,6 +1178,42 @@ class VariantDatasetFunctions(private val vds: VariantSampleMatrix[Genotype]) ex
     ret
   }
 
+
+  def queryGenotypes(expr: String): (Annotation, Type) = {
+    val qv = queryGenotypes(Array(expr))
+    assert(qv.length == 1)
+    qv.head
+  }
+
+  def queryGenotypes(exprs: Array[String]): Array[(Annotation, Type)] = {
+    val aggregationST = Map(
+      "global" -> (0, vds.globalSignature),
+      "g" -> (1, TGenotype),
+      "v" -> (2, TVariant),
+      "va" -> (3, vds.vaSignature),
+      "s" -> (4, TSample),
+      "sa" -> (5, vds.saSignature))
+    val ec = EvalContext(Map(
+      "global" -> (0, vds.globalSignature),
+      "gs" -> (1, TAggregable(TGenotype, aggregationST))))
+
+    val ts = exprs.map(e => Parser.parseExpr(e, ec))
+
+    val localGlobalAnnotation = vds.globalAnnotation
+    val (zVal, seqOp, combOp, resOp) = Aggregators.makeFunctions[(Genotype, Variant, Annotation, String, Annotation)](ec,
+      { case (ec, (g, v, va, s, sa)) =>
+      ec.setAll(localGlobalAnnotation, g, v, va, s, sa)
+    })
+
+    val result = vds.expandWithAll()
+        .map { case (v, va, s, sa, g) => (g, v, va, s, sa) }
```

Comment:
Can you make the order of the types in makeFunctions match expandWithAll so you don't need to do this?  It allocates a 5-tuple for every genotype!  SAD.